spotify_player/src/config/mod.rs

```diff
@@ -303,14 +331,98 @@ impl Default for DeviceConfig {
     }
 }
 
+impl Default for LayoutConfig {
+    fn default() -> Self {
+        Self {
+            library: LibraryLayoutConfig {
+                playlist_percent: 40,
+                album_percent: 40,
+            },
+            search: SearchLayoutConfig {
+                top_percent: 50,
+                left_percent: 50,
+            },
+            artist: ArtistLayoutConfig {
+                album_percent: 50,
+                top_songs_percent: 60,
+            },
+            playback_window_position: Position::Top,
+            playback_window_height: 6,
+        }
+    }
+}
+
+impl CheckValues for LayoutConfig {
+    fn check_values(&self) -> Result<(), String> {
+        let mut err_string = String::new();
+
+        err_string = LayoutConfig::concat_errors(err_string, &self.library.check_values());
+
+        err_string = LayoutConfig::concat_errors(err_string, &self.search.check_values());
+
+        err_string = LayoutConfig::concat_errors(err_string, &self.artist.check_values());
+
+        if err_string.is_empty() {
+            return Ok(());
+        }
+
+        Err(err_string)
+    }
+}
+
+impl LayoutConfig {
+    fn concat_errors(prev_msgs: String, new_msg: &Result<(), String>) -> String {
+        if let Err(e) = new_msg {
+            return format!("{} \n {}", prev_msgs, e);
+        }
+        prev_msgs
+    }
+}
+
+impl CheckValues for LibraryLayoutConfig {
+    fn check_values(&self) -> Result<(), String> {
+        if self.album_percent + self.playlist_percent > 99 {
+            return Err(
+                "Library-Layout album_percent and playlist_percent cannot be greater than 99!"
+                    .to_string(),
+            );
+        }
+        Ok(())
+    }
+}
+
+impl CheckValues for SearchLayoutConfig {
+    fn check_values(&self) -> Result<(), String> {
+        if self.top_percent > 99 || self.left_percent > 99 {
+            return Err("Search-Layout options are invalid!".to_string());
+        }
+        Ok(())
+    }
+}
+
+impl CheckValues for ArtistLayoutConfig {
+    fn check_values(&self) -> Result<(), String> {
+        if self.album_percent > 99 || self.top_songs_percent > 99 {
+            return Err("Artist-Layout options are invalid!".to_string());
+        }
+        Ok(())
+    }
+}
```

Comment:
this introduces a bit of code boilerplates. You can implement all the checks inside a single `Layout::check_values` function, e.g

```
if self.album_percent + self.playlist_percent > 99 {
     anyhow::bail!("invalid library layout: summation of album_percent and playlist_percent cannot be greater than 99!");
}
```

Reply:
the `check_values` function can return [`anyhow::Error`](https://docs.rs/anyhow/latest/anyhow/struct.Error.html) instead